src/main/java/com/google/opensesame/servlets/UserServlet.java

```diff
@@ -73,58 +58,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
+    String[] interestTagsParam = request.getParameterValues("interestTags");
+    if (interestTagsParam == null) {
+      interestTagsParam = new String[] {};
+    }
+    ArrayList<String> userTags = new ArrayList<>(Arrays.asList(interestTagsParam));
 
-    Entity personEntity =
-        new PersonBuilder().gitHubID(userGitHubID).interestTags(userTagsList).buildPersonEntity();
-    datastore.put(personEntity);
-  }
+    User user = AuthServlet.getAuthorizedUser();
+    if (user == null) {
+      ErrorResponse.sendJsonError(
+          response,
+          "User not logged in",
+          HttpServletResponse.SC_FORBIDDEN,
+          "You are not logged in");
+      return;
+    }
 
-  // TODO: use function in other servlets.
-  /**
-   * Query datastore for entities. Comparisons are done with field coming first. For example field
-   * EQUAL value. Look at
-   * https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Query.FilterOperator
-   * for more information on the operator
-   *
-   * @param EntityName the type of entity to be queried for
-   * @param field the field that is being used to query
-   * @param value the object that will be compared to @param field
-   * @param operator the type of comparison to be made.
-   * @return {PreparedQuery} a Datastore prepared query.
-   */
-  public PreparedQuery queryInDatabase(
-      String entityName, String field, Object value, FilterOperator operator) {
-    Filter userFilter = new FilterPredicate(field, operator, value);
-    return datastore.prepare(new Query(entityName).setFilter(userFilter));
-  }
+    String userId = user.getUserId();
 
-  // TODO: make error handling conform with Richie's error handling
-  // TODO: update to Objectify
-  /**
-   * Check if a user is stored with userID in Datastore. Invalid return false.
-   *
-   * @param userID
-   * @return true if userID exists in datastore, false otherwise.
-   */
-  public Boolean hasProfile(String userID) {
-    Key userKey = KeyFactory.createKey(PersonBuilder.ENTITY_NAME, userID);
+    // Get User information from GitHub using the Oath token.
+    GHMyself userGHMyself;
     try {
-      datastore.get(userKey);
-    } catch (EntityNotFoundException e) {
-      return false;
+      userGHMyself = GitHub.connectUsingOAuth(userGitHubAuthToken).getMyself();
+    } catch (Exception e) {
+      ErrorResponse.sendJsonError(
+          response,
+          e.getMessage()
+              + e.getStackTrace()
+              + "Could not get authenticated User with the given token",
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "User could not be authenticated by GitHub, please try again");
+      return;
     }
-    return true;
+    // Build and send the User's datastore entity
+    ofy().save().entity(new UserEntity(userId, userGHMyself.getLogin(), userTags, user.getEmail()));
+    // TODO: add check for hasProfile.
   }
+  // TODO: make function to get users
```

Comment:
I think there is already a function to get a user based on ID. Either this TODO is already completed or it should be more specific.